dbms/src/Common/getNumberOfLogicalCPUCores.cpp

```diff
@@ -0,0 +1,125 @@
+// Copyright 2022 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <Common/Exception.h>
+#include <Common/getNumberOfLogicalCPUCores.h>
+
+#include <thread>
+
+#if defined(__linux__)
+#include <cmath>
+#include <fstream>
+#endif // __linux__
+
+namespace DB::ErrorCodes
+{
+extern const int CPUID_ERROR;
+} // namespace DB::ErrorCodes
+
+#if defined(__linux__)
+// Try to look at cgroups limit if it is available.
+static inline int read_int_from(const char * filename, int default_value)
+{
+    std::ifstream infile(filename);
```

Comment:
nit: use poco lib to read file?

Reply:
`Poco::FileInputStream` will throw an exception when file no found directly.